interpreter/text/parser.mly

```diff
@@ -249,15 +265,20 @@ func_type :
     { let FuncType (ins, out) = $6 in FuncType ($4 :: ins, out) }
 
 table_type :
-  | limits elem_type { TableType ($1, $2) }
+  | limits32 elem_type { TableType ($1, $2) }
```

Comment:
This doesn't seem to work. The use of `nat32` binds parameter `nat` to always have a 32-bit type, and then the calls with `nat64` fail (or, if change the order of `table_type` and `memory_type` then the nat32 fails.. not impressed with this "type inference" ;)

This can probably be fixed by giving `fun nat` an explicit polymorphic signature but that will make things rather unreadable. Suggestions?

Reply:
```
Error: This expression has type string -> Source.region -> I64.t
       but an expression was expected of type
         string -> Source.region -> I32.t
```

Reply:
Ah, yacc probably emits a bunch of mutually recursive functions, and type inference does not allow polymorphic recursion without annotation (because that's undecidable ;) ). There probably is no easy fix, so fine to leave it as is.